docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,271 @@
+2. Need to use different volume types than the ones present in the `CloudProfile`.
+3. Extending the expiration date of Kubernetes versions. Given that the `CloudProfile` is a cluster-scoped resource, it is currently not possible to extend the expiration date for shoots in one project but only centrally for all projects.
+4. Extending the expiration date for machine images. For the same reasons as extending the expiration date of the Kubernetes versions.
+
+### Current State
+
+`CloudProfile`s are non-namespaced resources. This means that in a typical Gardener installation, only a handful of `CloudProfile`s (typically one per cloud provider) exist. They can be consumed by any shoot cluster. Consequently, when a project requires changes to `CloudProfile`s for any of the reasons mentioned in the [context section](#context), they are changed for the entire Gardener landscape. Since some projects might require frequent changes, this becomes quite a cumbersome process on both, the operators' and users' sides.
+
+### Goals
+
+- Reduce load on Gardener operators for maintaining `CloudProfile`s
+- Possibly reduce the time a team has to wait until a change in a `CloudProfile` is reflected in the landscape.
+- Make it so that project-scoped information in the `CloudProfile` is only visible to the relevant project
+- Full backward compatibility
+
+### Non-Goals
+
+- Automate the approval process for changes in `CloudProfile`s
+- Add different, possibly unsupported Kubernetes versions or machine image versions and names
+
+## Proposal
+
+It is proposed to implement a solution that enables project administrators to create custom (private) `CloudProfile`s. These `CloudProfile`s would be consumed by project users of the project they were created in.
+
+### Approach
+
+First of all, a new, namespaced API object `PrivateCloudProfile` is defined. Its type definition is very similar to the `CloudProfile` object, but omits the following fields:
+
+The general approach is that a `PrivateCloudProfile` inherits from a `CloudProfile` using a `parent` field. Fields such as `machineTypes`, `volumeTypes` and `caBundle` are going to be merged with the parent `CloudProfile`. However, a restriction needs to be defined so that the `kubernetes` and `machineImages` fields in a `PrivateCloudProfile` may only be adjusted by a Gardener operator to reduce the chance of a team staying on an unsupported Kubernetes version. A similar problem is already solved in Gardener using custom RBAC verbs [here](https://github.com/gardener/gardener/blob/master/plugin/pkg/global/customverbauthorizer), see [custom RBAC verb section](#custom-rbac-verb) for more information.
+
+Currently, the shoot's reference to a `CloudProfile` is immutable. This validation will be relaxed to allow updating to a `PrivateCloudProfile` whose parent is the same as the currently configured `CloudProfile`. The change will also be reversible, i.e. switching from `PrivateCloudProfile` to `CloudProfile`.
```

Comment:
Let's assume for a moment a totally independent on-prem managed infrastructure like OpenStack or vSphere. It makes little/no sense to inherit from any of our current cloud profiles. Let's assume bare metal / manual provisioning. Here, it makes even less sense to inherit from any of our current cloud profiles. What's our play then?

Maybe introducing a minimal cluster-scoped root cloud profile from which to inherit, so that we can at least control the Kubernetes versions, which we actually have to control because those are built-in into Gardener? This cluster-scoped root cloud profile could then also become the parent for the existing cluster-scoped cloud profiles, if we allow multi-level inheritance (deduplicating the Kubernetes versions, which should probably never have been duplicated into the cluster-scoped cloud profiles in the first place).

Reply:
Still defined out of scope for now. However, the proposal was adjusted to potentially cover this use-case in the future.